### ISSUE TITLE
allow_embedding

### ISSUE THREAD

**Issue by gieljnssns** [thorrak/fermentrack#571]
I'm trying to embed my dashboard (http://192.168.79.219/devices/4/dashboard/) in home-assistant as an [iframe](https://www.home-assistant.io/integrations/panel_iframe/).
But when I open the page, my dashboard doesn't load
This is the error in the console
```
Refused to display 'http://192.168.79.219/devices/4/dashboard/' in a frame because it set 'X-Frame-Options' to 'deny'.
```
I've searched a wile and found that there must be a setting to `allow_embedding = true` [link](https://community.home-assistant.io/t/https-grafana-panel-does-not-show-up/125321).
Is there such a setting in Fermentrack?

**Comment by mp-se** [comment]
This is a security feature built into django and the easy way to disable it for the whole site is to do that in the web server configuration. 

login to the computer with root privileges

go to the directory /etc/nginx/sites-enabled where the fermentrack web configuration is stored.

edit the file default-fermentrack

update the following section with the last option "proxy_hide_header" that will remove the header that is blocking the iframe.

    location @applocation {                                                                                                                                                                                 
      proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;                                                                                                                                          
      proxy_set_header Host $http_host;                                                                                                                                                                     
      proxy_redirect off;                                                                                                                                                                                   
      proxy_pass http://127.0.0.1:8123;                                                                                                                                                                     
      proxy_hide_header X-Frame-Options;                                                                                                                                                                    
    }

**Comment by gieljnssns** [comment]
Thanks a lot that did it

afterwards `sudo systemctl restart nginx`

For me this can be closed, but maybe can this made easier in the admin panel?

**Comment by gieljnssns** [comment]
And another question, is it possible to only use this
<img width="311" alt="Schermafbeelding 2021-01-05 om 06 35 33" src="https://user-images.githubusercontent.com/17234951/103610339-3c0d8980-4f20-11eb-947f-d54293d2a8df.png">
as an iframe?

**Comment by mp-se** [comment]
You can get the data via the API but you need to handle rendering in home assistant.  This will get the data for lcd 1. You can probably use the REST API integration to extract the data and create your own panel via yaml. 

http://<your ip>/api/lcd/1/

**Comment by mp-se** [comment]
> Thanks a lot that did it
> 
> afterwards `sudo systemctl restart nginx`
> 
> For me this can be closed, but maybe can this made easier in the admin panel?

I dont think so, bypassing this in django requires some development and removing security is probably not a good thing. You cannot manage the nginx server from the django control panel.

**Comment by gieljnssns** [comment]
OK thanks

**Comment by gieljnssns** [comment]
@mp-se 
Just another smal question inn the same range...
When I do - Add to homescreen - on my iPhone, I want `http://192.168.79.219/devices/4/dashboard/` to show as a web app, I mean without url on top or forward and back switches at the bottom.
Is this also possible?

**Comment by gieljnssns** [comment]
Now it looks like this
![IMG_4968](https://user-images.githubusercontent.com/17234951/103642465-76444e80-4f53-11eb-976d-965e01e388ad.PNG)

And I want it like this (also add to homescreen)
![IMG_4967](https://user-images.githubusercontent.com/17234951/103642543-8f4cff80-4f53-11eb-81d5-6694b2ec132a.PNG)

**Comment by mp-se** [comment]
Im not a IOS developer but I dont think that is possible without creating an application for IOS.

**Comment by thorrak** [comment]
> Now it looks like this
> ![IMG_4968](https://user-images.githubusercontent.com/17234951/103642465-76444e80-4f53-11eb-976d-965e01e388ad.PNG)
> 
> And I want it like this (also add to homescreen)
> ![IMG_4967](https://user-images.githubusercontent.com/17234951/103642543-8f4cff80-4f53-11eb-81d5-6694b2ec132a.PNG)

@Garzlok did some work to enable this, but I haven’t merged it in, as it’s not really the model Fermentrack was built to implement - and is an iOS specific change. 

I can take a look to see if this could be done via a Constance flag that would be set in the admin, but if that doesn’t work, I don’t see this getting added in the near future.

**Comment by mcinnes01** [comment]
I'm working on a hassio addon for fermentrack, allowing it be installed through the addon store and accessed in home assistant.

I have 2 outstanding issues to get it fully working:
- I need to be able to specify the location of the /data /log /db folders used by fermentrack 
- The other I think may have been answer or possibly touched on in this thread around the proxy_hide_header
